scripts/imgtool/image.py

```diff
@@ -612,6 +647,21 @@ def create(self, key, public_key_format, enckey, dependencies=None,
             if compression_tlvs is not None:
                 for tag, value in compression_tlvs.items():
                     prot_tlv.add(tag, value)
+
+            if self.vid is not None:
+                vid = parse_uuid(uuid.NAMESPACE_DNS, self.vid)
+                payload = struct.pack(e + '16s', vid)
+                prot_tlv.add('UUID_VID', payload)
+
+            if self.cid is not None:
+                if self.vid is not None:
+                    namespace = uuid.UUID(bytes=vid)
+                else:
+                    namespace = None
```

Comment:
why does the above use DNS and this either use the above or nothing?

Reply:
The `None` option is valid if and only if the user provides RAW CID UUID value.

Reply:
It also aligns with the recommendations in SUIT specification:
> If the Vendor ID is a UUID, the RECOMMENDED method to create a Class ID is:
> Class ID = UUID5(Vendor ID, Class-Specific-Information)